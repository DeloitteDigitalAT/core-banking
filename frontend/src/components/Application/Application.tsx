import React, { useState } from "react";
import { Accordion, Documents, Form, Header, SmallButton } from "../UI";
import img1 from "../../assets/images/Id.jpeg";
import img2 from "../../assets/images/Payslip.jpg";
import img3 from "../../assets/images/Meldezettel.png";

const mockData = [
  {
    client_id: 1,
    created_date: "2021-03-08",
    first_name: "Judy",
    last_name: "Green",
    date_of_birth: "1970-11-03",
    social_security_number: "377-82-5154",
    gender: "F",
    marital_status: "Engaged",
    residential_status: "With Family",
    citizenship: "Turkmenistan",
    address_details_id: 8,
    house_no: 701,
    street_name: "898 Cynthia Inlet",
    city: "Graz",
    postal_code: 1200,
    phone_number: "(930)544-3449x587",
    mobile_number: null,
    email_id: "judy.green@gmail.com",
    employment_status: "Selfemployed",
    annual_income: 84000.0,
    employed_years: 17,
    position: "Selfemployed",
    company_employed: "N/A",
    credit_amount: 76000.0,
    interest_rate: 0.68,
    planned_emi: 3209.73,
    duration_in_months: 24,
    files: [
      {
        document_list_id: 1,
        name: "ID",
        description: "Identity Card",
        location_url: img1,
        category: "ID",
        created_date: "2021-03-08",
      },
      {
        document_list_id: 21,
        name: "Payslip1",
        description: "Payslip for month 1",
        location_url: img2,
        category: "Payslip",
        created_date: "2021-03-08",
      },
      {
        document_list_id: 41,
        name: "Payslip2",
        description: "Payslip for month 2",
        location_url: img2,
        category: "Payslip",
        created_date: "2021-03-08",
      },
      {
        document_list_id: 61,
        name: "Payslip3",
        description: "Payslip for month 3",
        location_url: img2,
        category: "Payslip",
        created_date: "2021-03-08",
      },
      {
        document_list_id: 81,
        name: "Meldezettel",
        description: "Address proof",
        location_url: img3,
        category: "Meldezettel",
        created_date: "2021-03-08",
      },
    ],
  },
];

interface ApplicationType {
  applicationID: string;
}

const Application: React.FC<ApplicationType> = properties => {
  const { applicationID } = properties;
  const application = mockData[0];
  const [personalData, setPersonalData] = useState([
    {
      id: 1,
      type: "input",
      label: "First Name",
      placeholder: "",
      value: application.first_name,
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6 md:col-span-3",
    },
    {
      id: 2,
      type: "input",
      label: "Last Name",
      placeholder: "",
      value: application.last_name,
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6 md:col-span-3 md:pl-4",
    },
    {
      id: 3,
      type: "input",
      label: "Date of birth",
      placeholder: "",
      value: application.date_of_birth,
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-2 md:col-start-1 col-span-1",
    },
    {
      id: 4,
      type: "input",
      label: "Gender",
      placeholder: "",
      value: application.gender,
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-1 pl-4",
    },
    {
      id: 5,
      type: "input",
      label: "Social security number",
      placeholder: "",
      value: application.social_security_number ? `${application.social_security_number}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3 pl-4",
    },
    {
      id: 6,
      type: "input",
      label: "Citizenship",
      placeholder: "",
      value: application.citizenship ? `${application.citizenship}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6",
    },
    {
      id: 7,
      type: "input",
      label: "Marital status",
      placeholder: "",
      value: application.marital_status ? `${application.marital_status}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3",
    },
    {
      id: 8,
      type: "input",
      label: "Residential status",
      placeholder: "",
      value: application.residential_status ? `${application.residential_status}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3 pl-4",
    },
  ]);
  const [address, setAddress] = useState([
    {
      id: 1,
      type: "input",
      label: "Address",
      placeholder: "",
      value: `${application.postal_code}, ${application.city}, ${application.street_name}, ${application.house_no}`,
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6",
    },
    {
      id: 2,
      type: "input",
      label: "Phone number",
      placeholder: "",
      value: application.phone_number ? `${application.phone_number}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6 md:col-span-2",
    },
    {
      id: 3,
      type: "input",
      label: "Mobile number",
      placeholder: "",
      value: application.mobile_number ? `${application.mobile_number}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6 md:col-span-2 md:pl-4",
    },
    {
      id: 4,
      type: "input",
      label: "Email",
      placeholder: "",
      value: application.email_id ? `${application.email_id}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6 md:col-span-2 md:pl-4",
    },
  ]);
  const [employmentStatus, setEmploymentStatus] = useState([
    {
      id: 1,
      type: "input",
      label: "Employment status",
      placeholder: "",
      value: application.employment_status ? `${application.employment_status}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6 md:col-span-2",
    },
    {
      id: 2,
      type: "input",
      label: "Annual income (Euro)",
      placeholder: "",
      value: application.annual_income ? `${application.annual_income}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3 md:col-span-2 md:pl-4",
    },
    {
      id: 3,
      type: "input",
      label: "Employed years",
      placeholder: "",
      value: application.employed_years ? `${application.employed_years}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3 md:col-span-2 pl-4",
    },
    {
      id: 4,
      type: "input",
      label: "Position",
      placeholder: "",
      value: application.position ? `${application.position}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6 md:col-span-3",
    },
    {
      id: 5,
      type: "input",
      label: "Company employed",
      placeholder: "",
      value: application.company_employed ? `${application.company_employed}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-6 md:col-span-3 md:pl-4",
    },
  ]);
  const [creditInfo, setCreditInfo] = useState([
    {
      id: 1,
      type: "input",
      label: "Credit amount requested",
      placeholder: "",
      value: application.credit_amount ? `${application.credit_amount}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3 md:col-span-2",
    },
    {
      id: 2,
      type: "input",
      label: "Interest rate",
      placeholder: "",
      value: application.interest_rate ? `${application.interest_rate}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3 md:col-span-1 pl-4",
    },
    {
      id: 3,
      type: "input",
      label: "Planned emi",
      placeholder: "",
      value: application.planned_emi ? `${application.planned_emi}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3 md:col-span-1 md:pl-4",
    },
    {
      id: 4,
      type: "input",
      label: "Duration in months",
      placeholder: "",
      value: application.duration_in_months ? `${application.duration_in_months}` : "",
      required: false,
      readonly: true,
      error: false,
      divClass: "col-span-3 md:col-span-2 pl-4",
    },
  ]);
  const [openAll, setOpenAll] = useState(1);
  return (
    <div className="flex-grow mb-4">
      <Header>Application ID: {applicationID}</Header>
      <div className="grid grid-cols-6 mb-2 text-sm">
        <div className="col-span-3">Creation date: {application.created_date}</div>
        <div className="col-span-3 text-right">
          <SmallButton text="open all" handler={() => setOpenAll(openAll > 0 ? openAll + 1 : openAll * -1)} active />
          <SmallButton
            text="close all"
            handler={() => setOpenAll(openAll > 0 ? openAll * -1 : openAll - 1)}
            active
            extraClass="ml-2"
          />
        </div>
      </div>
      <Accordion label="Personal data" color="pacific" open={openAll}>
        <Form elements={personalData} setForm={() => setPersonalData} />
      </Accordion>
      <Accordion label="Address" color="pacific" open={openAll}>
        <Form elements={address} setForm={() => setAddress} />
      </Accordion>
      <Accordion label="Employment status" color="pacific" open={openAll}>
        <Form elements={employmentStatus} setForm={() => setEmploymentStatus} />
      </Accordion>
      <Accordion label="Credit information" color="pacific" open={openAll}>
        <Form elements={creditInfo} setForm={() => setCreditInfo} />
      </Accordion>
      <Accordion label="Documents" color="pacific" open={openAll}>
        <Documents documents={application.files} />
      </Accordion>
    </div>
  );
};

export default Application;
