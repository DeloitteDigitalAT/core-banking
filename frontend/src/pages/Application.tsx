import React from "react";
import { RouteComponentProps } from "react-router-dom";
import Application from "../components/Application/Application";

interface MatchParams {
  applicationID: string;
}

const PageApplication: React.FC<RouteComponentProps<MatchParams>> = properties => {
  const { applicationID } = properties.match.params;
  return <Application applicationID={applicationID} />;
};

export default PageApplication;
