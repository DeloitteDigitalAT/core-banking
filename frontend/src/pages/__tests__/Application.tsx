import React from "react";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import Application from "../Application";

const history = createMemoryHistory();
const match = { params: { applicationID: "123" }, isExact: true, path: "/", url: "" };
const location = { pathname: "/", search: "", hash: "", state: undefined, key: "1" };

describe("UI/Application page", () => {
  test("snapshot renders", () => {
    const tree = renderer.create(<Application match={match} history={history} location={location} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
