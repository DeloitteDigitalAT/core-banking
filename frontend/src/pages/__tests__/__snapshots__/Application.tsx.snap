// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Application page snapshot renders 1`] = `
<div
  className="flex-grow mb-4"
>
  <div
    className="text-xl py-4 mb-0 text-gray-800"
  >
    Application ID: 
    123
  </div>
  <div
    className="grid grid-cols-6 mb-2 text-sm"
  >
    <div
      className="col-span-3"
    >
      Creation date: 
      2021-03-08
    </div>
    <div
      className="col-span-3 text-right"
    >
      <button
        className="bg-gradient-to-b from-green-100 to-green-200 border-green-100 text-xs hover:bg-green-400 hover:from-green-200 hover:to-green-300 border text-green-800 rounded-md px-2 py-0 transition duration-500 ease select-none focus:outline-none focus:shadow-outline undefined"
        onClick={[Function]}
        type="button"
      >
        open all
      </button>
      <button
        className="bg-gradient-to-b from-green-100 to-green-200 border-green-100 text-xs hover:bg-green-400 hover:from-green-200 hover:to-green-300 border text-green-800 rounded-md px-2 py-0 transition duration-500 ease select-none focus:outline-none focus:shadow-outline ml-2"
        onClick={[Function]}
        type="button"
      >
        close all
      </button>
    </div>
  </div>
  <div>
    <div
      className="grid grid-cols-6 -mx-4 border-l-2 block bg-opacity-30 uppercase tracking-wide text-xs font-semibold text-green-800 pl-4
        py-2 cursor-pointer border-pacific-200 bg-gradient-to-r from-pacific-100 to-pacific-50 hover:from-pacific-200 hover:to-pacific-100"
      onClick={[Function]}
    >
      <div
        className="col-span-5 pt-1"
      >
        Personal data
      </div>
      <div
        className="col-span-1 w-6 h-6 justify-center items-center justify-self-end mr-4 flex text-pacific-400"
      >
        <svg
          className="h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="pl-4 pr-4 -mx-4 border-l-2  overflow-hidden pb-6 border-pacific-100"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "auto",
          "maxHeight": "10000px",
          "paddingBottom": "1.5rem",
          "transition": "max-height 1s",
        }
      }
    >
      <div
        className="grid grid-cols-6"
      >
        <div
          className="col-span-6 md:col-span-3"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              First Name
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="Judy"
            />
          </label>
        </div>
        <div
          className="col-span-6 md:col-span-3 md:pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Last Name
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="Green"
            />
          </label>
        </div>
        <div
          className="col-span-2 md:col-start-1 col-span-1"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Date of birth
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="1970-11-03"
            />
          </label>
        </div>
        <div
          className="col-span-1 pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Gender
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="F"
            />
          </label>
        </div>
        <div
          className="col-span-3 pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Social security number
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="377-82-5154"
            />
          </label>
        </div>
        <div
          className="col-span-6"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Citizenship
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="Turkmenistan"
            />
          </label>
        </div>
        <div
          className="col-span-3"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Marital status
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="Engaged"
            />
          </label>
        </div>
        <div
          className="col-span-3 pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Residential status
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="With Family"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="grid grid-cols-6 -mx-4 border-l-2 block bg-opacity-30 uppercase tracking-wide text-xs font-semibold text-green-800 pl-4
        py-2 cursor-pointer border-pacific-200 bg-gradient-to-r from-pacific-100 to-pacific-50 hover:from-pacific-200 hover:to-pacific-100"
      onClick={[Function]}
    >
      <div
        className="col-span-5 pt-1"
      >
        Address
      </div>
      <div
        className="col-span-1 w-6 h-6 justify-center items-center justify-self-end mr-4 flex text-pacific-400"
      >
        <svg
          className="h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="pl-4 pr-4 -mx-4 border-l-2  overflow-hidden pb-6 border-pacific-100"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "auto",
          "maxHeight": "10000px",
          "paddingBottom": "1.5rem",
          "transition": "max-height 1s",
        }
      }
    >
      <div
        className="grid grid-cols-6"
      >
        <div
          className="col-span-6"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Address
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="1200, Graz, 898 Cynthia Inlet, 701"
            />
          </label>
        </div>
        <div
          className="col-span-6 md:col-span-2"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Phone number
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="(930)544-3449x587"
            />
          </label>
        </div>
        <div
          className="col-span-6 md:col-span-2 md:pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Mobile number
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value=""
            />
          </label>
        </div>
        <div
          className="col-span-6 md:col-span-2 md:pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Email
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="judy.green@gmail.com"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="grid grid-cols-6 -mx-4 border-l-2 block bg-opacity-30 uppercase tracking-wide text-xs font-semibold text-green-800 pl-4
        py-2 cursor-pointer border-pacific-200 bg-gradient-to-r from-pacific-100 to-pacific-50 hover:from-pacific-200 hover:to-pacific-100"
      onClick={[Function]}
    >
      <div
        className="col-span-5 pt-1"
      >
        Employment status
      </div>
      <div
        className="col-span-1 w-6 h-6 justify-center items-center justify-self-end mr-4 flex text-pacific-400"
      >
        <svg
          className="h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="pl-4 pr-4 -mx-4 border-l-2  overflow-hidden pb-6 border-pacific-100"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "auto",
          "maxHeight": "10000px",
          "paddingBottom": "1.5rem",
          "transition": "max-height 1s",
        }
      }
    >
      <div
        className="grid grid-cols-6"
      >
        <div
          className="col-span-6 md:col-span-2"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Employment status
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="Selfemployed"
            />
          </label>
        </div>
        <div
          className="col-span-3 md:col-span-2 md:pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Annual income (Euro)
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="84000"
            />
          </label>
        </div>
        <div
          className="col-span-3 md:col-span-2 pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Employed years
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="17"
            />
          </label>
        </div>
        <div
          className="col-span-6 md:col-span-3"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Position
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="Selfemployed"
            />
          </label>
        </div>
        <div
          className="col-span-6 md:col-span-3 md:pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Company employed
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="N/A"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="grid grid-cols-6 -mx-4 border-l-2 block bg-opacity-30 uppercase tracking-wide text-xs font-semibold text-green-800 pl-4
        py-2 cursor-pointer border-pacific-200 bg-gradient-to-r from-pacific-100 to-pacific-50 hover:from-pacific-200 hover:to-pacific-100"
      onClick={[Function]}
    >
      <div
        className="col-span-5 pt-1"
      >
        Credit information
      </div>
      <div
        className="col-span-1 w-6 h-6 justify-center items-center justify-self-end mr-4 flex text-pacific-400"
      >
        <svg
          className="h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="pl-4 pr-4 -mx-4 border-l-2  overflow-hidden pb-6 border-pacific-100"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "auto",
          "maxHeight": "10000px",
          "paddingBottom": "1.5rem",
          "transition": "max-height 1s",
        }
      }
    >
      <div
        className="grid grid-cols-6"
      >
        <div
          className="col-span-3 md:col-span-2"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Credit amount requested
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="76000"
            />
          </label>
        </div>
        <div
          className="col-span-3 md:col-span-1 pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Interest rate
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="0.68"
            />
          </label>
        </div>
        <div
          className="col-span-3 md:col-span-1 md:pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Planned emi
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="3209.73"
            />
          </label>
        </div>
        <div
          className="col-span-3 md:col-span-2 pl-4"
        >
          <label
            className="block mt-4"
          >
            <span
              className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
            >
              Duration in months
            </span>
            <input
              className="form-input mt-1 block w-full text-sm border rounded py-2 px-2  outline-none"
              onChange={[Function]}
              placeholder=""
              readOnly={true}
              value="24"
            />
          </label>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="grid grid-cols-6 -mx-4 border-l-2 block bg-opacity-30 uppercase tracking-wide text-xs font-semibold text-green-800 pl-4
        py-2 cursor-pointer border-pacific-200 bg-gradient-to-r from-pacific-100 to-pacific-50 hover:from-pacific-200 hover:to-pacific-100"
      onClick={[Function]}
    >
      <div
        className="col-span-5 pt-1"
      >
        Documents
      </div>
      <div
        className="col-span-1 w-6 h-6 justify-center items-center justify-self-end mr-4 flex text-pacific-400"
      >
        <svg
          className="h-5 w-5"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
    <div
      className="pl-4 pr-4 -mx-4 border-l-2  overflow-hidden pb-6 border-pacific-100"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "auto",
          "maxHeight": "10000px",
          "paddingBottom": "1.5rem",
          "transition": "max-height 1s",
        }
      }
    >
      <div
        className="mt-4"
      >
        <table
          className="min-w-max w-full table-auto -mx-2"
        >
          <thead
            className="bg-pacific-100 text-green-800 uppercase text-xs"
          >
            <tr>
              <th
                className="py-2 px-2 text-left"
              >
                ID
              </th>
              <th
                className="py-2 px-2 text-left"
              >
                Name
              </th>
              <th
                className="py-2 px-2 text-left"
              >
                Description
              </th>
              <th
                className="py-2 px-2 text-left"
              >
                Category
              </th>
              <th
                className="py-2 px-2 text-left"
              >
                Created date
              </th>
            </tr>
          </thead>
          <tbody
            className="text-gray-600 text-sm"
          >
            <tr
              className="border-b border-gray-200 "
            >
              <td
                className="py-2 px-2"
              >
                1
              </td>
              <td
                className="py-2 px-2 text-denim-600 font-medium"
              >
                <a
                  href="Id.jpeg"
                  rel="noreferrer"
                  target="_blank"
                >
                  ID
                </a>
              </td>
              <td
                className="py-2 px-2"
              >
                Identity Card
              </td>
              <td
                className="py-2 px-2"
              >
                ID
              </td>
              <td
                className="py-2 px-2"
              >
                2021-03-08
              </td>
            </tr>
            <tr
              className="border-b border-gray-200 bg-pacific-50 bg-opacity-50"
            >
              <td
                className="py-2 px-2"
              >
                21
              </td>
              <td
                className="py-2 px-2 text-denim-600 font-medium"
              >
                <a
                  href="Payslip.jpg"
                  rel="noreferrer"
                  target="_blank"
                >
                  Payslip1
                </a>
              </td>
              <td
                className="py-2 px-2"
              >
                Payslip for month 1
              </td>
              <td
                className="py-2 px-2"
              >
                Payslip
              </td>
              <td
                className="py-2 px-2"
              >
                2021-03-08
              </td>
            </tr>
            <tr
              className="border-b border-gray-200 "
            >
              <td
                className="py-2 px-2"
              >
                41
              </td>
              <td
                className="py-2 px-2 text-denim-600 font-medium"
              >
                <a
                  href="Payslip.jpg"
                  rel="noreferrer"
                  target="_blank"
                >
                  Payslip2
                </a>
              </td>
              <td
                className="py-2 px-2"
              >
                Payslip for month 2
              </td>
              <td
                className="py-2 px-2"
              >
                Payslip
              </td>
              <td
                className="py-2 px-2"
              >
                2021-03-08
              </td>
            </tr>
            <tr
              className="border-b border-gray-200 bg-pacific-50 bg-opacity-50"
            >
              <td
                className="py-2 px-2"
              >
                61
              </td>
              <td
                className="py-2 px-2 text-denim-600 font-medium"
              >
                <a
                  href="Payslip.jpg"
                  rel="noreferrer"
                  target="_blank"
                >
                  Payslip3
                </a>
              </td>
              <td
                className="py-2 px-2"
              >
                Payslip for month 3
              </td>
              <td
                className="py-2 px-2"
              >
                Payslip
              </td>
              <td
                className="py-2 px-2"
              >
                2021-03-08
              </td>
            </tr>
            <tr
              className="border-b border-gray-200 "
            >
              <td
                className="py-2 px-2"
              >
                81
              </td>
              <td
                className="py-2 px-2 text-denim-600 font-medium"
              >
                <a
                  href="Meldezettel.png"
                  rel="noreferrer"
                  target="_blank"
                >
                  Meldezettel
                </a>
              </td>
              <td
                className="py-2 px-2"
              >
                Address proof
              </td>
              <td
                className="py-2 px-2"
              >
                Meldezettel
              </td>
              <td
                className="py-2 px-2"
              >
                2021-03-08
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
